cpp/tests/sort/sort_test.cpp

```diff
@@ -740,16 +741,98 @@ TYPED_TEST(Sort, ZeroSizedColumns)
 
 TYPED_TEST(Sort, WithListColumn)
 {
-  using T = int;
-  lists_column_wrapper<T> lc{{1, 2, 3}, {4, 5, 6}, {7, 8, 9}};
-  CUDF_EXPECT_THROW_MESSAGE(cudf::sort(table_view({lc})),
-                            "Cannot lexicographic compare a table with a LIST column");
-
-  std::vector<std::unique_ptr<cudf::column>> child_cols;
-  child_cols.push_back(lc.release());
-  structs_column_wrapper sc{std::move(child_cols), {1, 0, 1}};
-  CUDF_EXPECT_THROW_MESSAGE(cudf::sort(table_view({sc})),
-                            "Cannot lexicographic compare a table with a LIST column");
+  using T = TypeParam;
+  if (std::is_same_v<T, bool>) { GTEST_SKIP(); }
+
+  using lcw = cudf::test::lists_column_wrapper<T, int32_t>;
+  lcw col{
+    {{1, 2, 3}, {}, {4, 5}, {}, {0, 6, 0}},
+    {{1, 2, 3}, {}, {4, 5}, {}, {0, 6, 0}},
+    {{1, 2, 3}, {}, {4, 5}, {0, 6, 0}},
+    {{1, 2}, {3}, {4, 5}, {0, 6, 0}},
+    {{7, 8}, {}},
+    lcw{lcw{}, lcw{}, lcw{}},
+    lcw{lcw{}},
+    {lcw{10}},
+    lcw{},
+  };
+
+  auto expect = cudf::test::fixed_width_column_wrapper<cudf::size_type>{8, 6, 5, 3, 0, 1, 2, 4, 7};
+  auto result = cudf::sorted_order(cudf::table_view({col}));
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expect, *result);
+}
+
+TYPED_TEST(Sort, WithNullableListColumn)
+{
+  using T = TypeParam;
+  if (std::is_same_v<T, bool>) { GTEST_SKIP(); }
+
+  using lcw = cudf::test::lists_column_wrapper<T, int32_t>;
+  lcw col{
+    {{1, 2, 3}, {}, {4, 5}, {}, {0, 6, 0}},                                          // 0
+    {{{1, 2, 3}, {}, {4, 5}, {}, {0, 6, 0}}, cudf::test::iterators::nulls_at({3})},  // 1
+    {{1, 2, 3}, {}, {4, 5}, {0, 6, 0}},                                              // 2
+    {{1, 2}, {3}, {4, 5}, {0, 6, 0}},                                                // 3
+    {{1, 2}, {3}, {4, 5}, {{0, 6, 0}, cudf::test::iterators::nulls_at({0})}},        // 4
```

Comment:
Similarly, type alias or `using cudf::test::iterators::nulls_at` at the beginning of the file, then here just use `nulls_at()`.